datalad/utils.py

```diff
@@ -542,11 +542,29 @@ def rmtemp(f, *args, **kwargs):
             lgr.debug("Path %s does not exist, so can't be removed" % f)
             return
         lgr.log(5, "Removing temp file: %s" % f)
-        # Can also be a directory
-        if os.path.isdir(f):
-            rmtree(f, *args, **kwargs)
-        else:
-            unlink(f)
+        # we have issues with not-yet closed git-annex processes.
+        # but a timing issue and a conceptual test failure are two different
+        # things, hence spend a few attempts and a couple of seconds to
+        # let an issue resolve itself
+        removed = False
+        last_exc = None
+        for attempt in range(5):
+            try:
+                # Can also be a directory
+                if os.path.isdir(f):
+                    rmtree(f, *args, **kwargs)
+                else:
+                    unlink(f)
+                removed = True
+                break
+            except PermissionError as e:
+                lgr.warn(
+                    'Attempt %i to remove temp location failed: %s',
+                    attempt + 1, e)
+                last_exc = e
+                time.sleep(2)
+        if not removed:
+            raise last_exc
```

Comment:
note -- we have `try_multiple` and `@try_multiple_dec` , and I thought that we already use them:
```
datalad/utils.py:@try_multiple_dec
datalad/utils.py-def unlink(f):
datalad/utils.py-    """'Robust' unlink.  Would try multiple times
--
datalad/utils.py:@try_multiple_dec
datalad/utils.py-def _rmtree(*args, **kwargs):
datalad/utils.py-    """Just a helper to decorate shutil.rmtree.
```
so you just need to use "our" `_rmtree` and `unlink` here

Reply:
But it is using `rmtree` from `datalad.utils`, which is our `rmtree`, and it does use `_rmtree`. But it "just" doesn't work ;-)